src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -675,7 +675,12 @@ export class Kernel implements IKernel {
         if (this.kernelConnectionMetadata.kind !== 'connectToLiveKernel') {
             // Gather all of the startup code at one time and execute as one cell
             const startupCode = await this.gatherStartupCode(notebookDocument);
-            await this.executeSilently(startupCode);
+            // Execute each startup code one at a time, as its possible that some startup code could fail.
+            // If something fails, we don't want the rest to fail, this way
+            // A single request could fail and the rest will continue on.
+            await startupCode.map((code) =>
+                this.executeSilently([code]).catch((ex) => traceVerbose(`Failed to execute startup code`, ex))
```

Comment:
This makes the whole thing take a lot longer. I think the code should each be wrapped in a try/except block instead.

Reply:
It's also not clear you can split it this way. The startup code may need to be grouped together to run as each line is a separate entry.